game/card.go

```diff
@@ -125,6 +128,25 @@ var Cards = map[CardName]*Card{
 		IsCreature:    true,
 	},
 
+	/*
+		Return a Forest you control to its owner's hand: Untap target creature.
+		Activate this ability only once each turn.
+	*/
+	QuirionRanger: &Card{
+		ActivatedAbility: &Effect{
+			Cost: &Effect{
+				EffectType: ReturnToHand,
```

Comment:
Oh, I see what you mean by "cost target" now. Still, it's not really a target. How about instead of TargetType we call it something like, Selector. And then it seems that rather than an Action having stuff like a CostTarget, maybe the action should have a Cost, and one type of cost would be a return-to-hand cost? I'm thinking of implementing Daze, here, specifically. But this could also handle specifying whether to pay red or blue mana for an inside out, for example.

Reply:
A specific question - can a quirion ranger return a forest to pay a cost, when that forest has pro green?

Reply:
i think so, it's not a target

Reply:
I guess "CastingCost" and "Cost" are the same thing, really. So prob just name them both "cost" so the same object can be used both for a casting cost and an activated ability cost?